### PR TITLE
MRG Simpler regression example

### PR DESCRIPTION
**Pull request by jona-sassenhagen**
With the new dataset and the metadata, we can now provide a simpler example for the linear_regression function.

@TalLinzen @teonbrooks

### CODE REVIEW

File: examples/stats/plot_sensor_regression.py [hunk 1]
```diff
@@ -10,68 +10,38 @@
...
-    x.plot_topomap(ch_type='mag', scalings=1., size=1.5, vmax=np.max,
-                   units=units, times=np.linspace(0.1, 0.2, 5))
+# Load the data
+path = kiloword.data_path() + '/kword_metadata-epo.fif'
+epochs = mne.read_epochs(path)
+print(epochs.metadata.head())
 
+# Add intercept column
+df = pd.DataFrame(epochs.metadata)
+epochs.metadata = df.assign(Intercept=[1 for _ in epochs.events])
```
Comment (agramfort) [inline examples/stats/plot_sensor_regression.py]: I would just do:

epochs.metadata['Intercept'] = 1.

Comment (jona-sassenhagen) [reply examples/stats/plot_sensor_regression.py]: I know, I would too, if it worked :)

@choldgraf this does not work, I assume metadata is immutable or something?

**Comment by jona-sassenhagen** [comment]
Here is what it looks like:
https://6493-1301584-gh.circle-artifacts.com/0/tmp/circle-artifacts.Tvbczub/html/auto_examples/stats/plot_sensor_regression.html

**Comment by agramfort** [comment]
before we had topos see:
https://martinos.org/mne/stable/auto_examples/stats/plot_sensor_regression.html

can you add this back?

**Comment by jona-sassenhagen** [comment]
I would just use plot_joint (had it in my first iteration), I'm pretty sure it's the best way of visualising evoked-shaped regression coefficients. (The Dufau et al paper uses evoked images.) Should I use plot_joint, and only plot_joint?

**Comment by agramfort** [comment]
add plot_joint

**Comment by jona-sassenhagen** [comment]
https://6494-1301584-gh.circle-artifacts.com/0/tmp/circle-artifacts.kXq7Sax/html/auto_examples/stats/plot_sensor_regression.html

**Comment by jona-sassenhagen** [comment]
Btw in the long term, I would deprecate the epochs regression function in favour of any combination of

- an example for how to use mne.decoding.Vectorizer + any regression function instead
- a thin wrapper around mne.decoding.Vectorizer + any regression function, optimally with an sklearn API

@kingjr

**Comment by jona-sassenhagen** [comment]
So where do we get a +2 from, maybe @olafhauk can have a look, it's very short :)

File: examples/stats/plot_sensor_regression.py [hunk 2]
```diff
@@ -5,73 +5,43 @@
...
+coefficient (beta value) for each combination of sensor and timepoint. Example
+shows the regression coefficient; the t and p values are also calculated
+automatically.
+
+Here, we repeat a few of the analyses from Dufau et al. (2014) by accessing
+the metadata object, which contains word-level information about various
+psycholinguistically relevant features of the words for which we have EEG
+activity.
+
+For the general methodology, see e.g. Hauk et al. (2006). The time course of
```
Comment (larsoner) [inline examples/stats/plot_sensor_regression.py]: Better to do proper references here with `[1]_` and `.. [1]` in a `References` section

File: examples/stats/plot_sensor_regression.py [hunk 3]
```diff
@@ -5,73 +5,43 @@
...
-regression coefficient (beta value) for each combination of sensor
-and timepoint. Example also shows the T statistics and the associated
-p-values.
-
-Note that this example is for educational purposes and that the data used
-here do not contain any significant effect.
-
-(See Hauk et al. (2006). The time course of visual word recognition as
-revealed by linear regression analysis of ERP data. Neuroimage.)
+individually, resulting in an Evoked object which contains the regression
```
Comment (larsoner) [inline examples/stats/plot_sensor_regression.py]:
```
:class:`mne.Evoked`
```

File: examples/stats/plot_sensor_regression.py [hunk 4]
```diff
@@ -5,73 +5,43 @@
...
+coefficient (beta value) for each combination of sensor and timepoint. Example
+shows the regression coefficient; the t and p values are also calculated
+automatically.
+
+Here, we repeat a few of the analyses from Dufau et al. (2014) by accessing
```
Comment (larsoner) [inline examples/stats/plot_sensor_regression.py]: ref

**Comment by jona-sassenhagen** [comment]
Bigram Frequency and Concreteness are continuous, so I am not quite sure what you mean.
I guess if at all, one could add something like in the metadata tutorial, e.g., plot_compare_evoked with a few bins.

**Comment by jona-sassenhagen** [comment]
>one could add something like in the metadata tutorial, e.g., plot_compare_evoked with a few bins.

 Notably, for this I'd wait for #4755 to be merged. So maybe we can merge this one now?

**Comment by larsoner** [comment]
Thanks @jona-sassenhagen, feel free to update some example with a comparison if you find time once these PRs are merged

**Comment by jona-sassenhagen** [comment]
Thanks guys!

**Pull request merged**